shims/common/src/main/scala/org/apache/gluten/config/GlutenConfig.scala

```diff
@@ -33,9 +34,13 @@ case class GlutenNumaBindingInfo(
     totalCoreRange: Array[String] = null,
     numCoresPerExecutor: Int = -1) {}
 
-class GlutenConfig(conf: SQLConf) extends Logging {
+class GlutenConfig(sessionOpt: Option[SparkSession] = None) extends Logging {
```

Comment:
Do we have to change the default constructor?

Could we just add a new API like 

```scala
object GlutenConf {
  def get(session: SparkSession): GlutenConfig = new GlutenConfig(session.sessionState.conf)
}
```

Reply:
The original construct that pass SQLConf is a mistake.
SQLConf is immuable, we can't pass it as a parameter.
We should get it with session.sessionState.conf first. Otherwise, we should get it with SQLConf.get if we can't hold the spark session.
`session.sessionState.conf` have better performance than `SQLConf.get`. Please refer https://github.com/apache/spark/pull/48693

Anyway, we can't pass SQLConf as parameter or declare it as a variable. The two methods are invalid:
`def someFunction(conf: SQLConf) ...`
`val conf: SQLConf = ...`

Reply:
You can see the tests `Gluten Clickhouse CI` is failed!
But I created another PR with the wrong method, but the tests `Gluten Clickhouse CI` is success

Reply:
I guess there is a bug. `Gluten Clickhouse` + `celeborn` is success in the wrong way, but failed in the right way.